### PR TITLE
Add support for Olimex STM32-H103

### PR DESCRIPTION
**Pull request by jeanthom**

### CODE REVIEW

File: src/stm32f103/olimexstm32h103/config.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#endif
+
+#ifndef HAVE_LED
+#define HAVE_LED 1
+#endif
+#ifndef LED_GPIO_PORT
+#define LED_GPIO_PORT GPIOC
+#endif
+#ifndef LED_GPIO_PIN
+#define LED_GPIO_PIN  GPIO14
```
Comment (devanlai) [inline src/stm32f103/olimexstm32h103/config.h]: From the schematic, it looks like the LED is actually on PC14.

File: src/stm32f103/olimexstm32h103/config.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#endif
+
+/* Technically, there is a button on PB8, but the button is
+   also shorted to BOOT0, so it's not very useful for us to
+   sample PB8 on boot, since pulling it high will already
+   trigger the ROM serial bootloader and prevent us from
+   running anyways. */
+#ifndef HAVE_BUTTON
+#define HAVE_BUTTON 0
+#endif
```
Comment (devanlai) [inline src/stm32f103/olimexstm32h103/config.h]: It looks like this board has an active-low button on PA0 that might be useful to enter the bootloader.

File: src/stm32f103/olimexstm32h103/config.h [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+#define USB_PULLUP_GPIO_PORT GPIOC
+#endif
+#ifndef USB_PULLUP_GPIO_PIN
+#define USB_PULLUP_GPIO_PIN  GPIO11
+#endif
+#ifndef USB_PULLUP_ACTIVE_HIGH
+#define USB_PULLUP_ACTIVE_HIGH 0
+#endif
+#ifndef USB_PULLUP_OPEN_DRAIN
+#define USB_PULLUP_OPEN_DRAIN 0
```
Comment (devanlai) [inline src/stm32f103/olimexstm32h103/config.h]: It looks like the Maple Mini and the STM32-H103 use the same USB pullup circuit, so for consistency with the Maple Mini configuration, I would prefer to configure this pin as open-drain. I don't think it makes any functional difference, but that's how the original LeafLabs Maple bootloader configured it and I think everyone else has followed their lead.

**Review by devanlai — CHANGES_REQUESTED** [review]
Hi @jeanthom, thanks for contributing a configuration for the STM32-H103. I don't have a board to test with, but based on the schematic, it looks like there are some minor tweaks that could be made.

I'd appreciate it if you could test the bootloader with those tweaks and confirm that everything still works.

File: src/stm32f103/olimexstm32h103/config.h (hunk 2 above)
Comment (jeanthom) [reply src/stm32f103/olimexstm32h103/config.h]: Fixed

File: src/stm32f103/olimexstm32h103/config.h (hunk 1 above)
Comment (jeanthom) [reply src/stm32f103/olimexstm32h103/config.h]: Fixed, but I'm not sure how to get the LED flashing to test my changes?

File: src/stm32f103/olimexstm32h103/config.h (hunk 3 above)
Comment (jeanthom) [reply src/stm32f103/olimexstm32h103/config.h]: Fixed

**Comment by jeanthom** [comment]
Hi, thanks for your feedback. I tried to fix everything, but for the LED I'm not sure how to test it? I haven't tried uploading firmwares yet, just made sure that I could enter the bootloader with the button or via the RTC registers and that it enumerated properly.

**Comment by devanlai** [comment]
Upon closer inspection, the bootloader doesn't actually do anything with the LED other than making sure it's off, so it doesn't make much of a difference, though maybe at some point in the future I'll make it blink the LED or something. I'm happy with these changes, so I'll go ahead and merge it now.

**Review by devanlai — APPROVED** [review]

**Comment by jeanthom** [comment]
Thanks!

**Pull request merged**